pkgs/native_assets_cli/lib/src/architecture.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi' show Abi;
+import 'dart:io';
+
+import 'os.dart';
+
+/// A hardware architecture which the Dart VM can run on.
+final class Architecture {
+  /// This architecture as used in [Platform.version].
+  final String dartPlatform;
+
+  const Architecture._(this.dartPlatform);
+
+  factory Architecture.fromAbi(Abi abi) => _abiToArch[abi]!;
```

Comment:
doc comment missing

Reply:
I tend to omit comments on methods that are self-explanatory (because the comment would be redundant - it says the same thing as the name and argument). Asked my desk mate and he tends to agree.

But I'm also fine adding it everywhere.

Done.

Reply:
I'd argue against adding redundant comments too.
So make a non-redundant comment and actually say something.

*How* is the `Abi` mapped to an `Architecture`? 
Can it refer to the `Abi` documentation which should describe itself as being an architecture specific calling convention? (Is it?)